includes/Core/Storage/Setting_With_Owned_Keys_Trait.php

```diff
@@ -51,21 +49,53 @@ function ( $option, $value ) use ( $keys ) {
 
 		add_filter(
 			'pre_update_option_' . static::OPTION,
-			function( $value, $old_value ) use ( $keys ) {
+			function ( $value, $old_value ) {
 				if ( is_array( $value ) && is_array( $old_value ) ) {
-					foreach ( $keys as $key ) {
-						if ( isset( $value[ $key ], $old_value[ $key ] ) && $value[ $key ] !== $old_value[ $key ] && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
-							$value['ownerID'] = get_current_user_id();
-							break;
-						}
-					}
+					return $this->maybe_update_owner_id_in_settings_before_saving_them( $value, $old_value );
 				}
-
-				return $value;
 			},
 			10,
 			2
 		);
 	}
 
+	/**
+	 * Updates settings to set the current user as an owner of the module if settings contain owned keys.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The module settings.
+	 */
+	protected function maybe_set_owner_id( $settings ) {
+		$keys = $this->get_owned_keys();
+		if ( count( array_intersect( array_keys( $settings ), $keys ) ) > 0 && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+			$this->merge( array( 'ownerID' => get_current_user_id() ) );
+		}
```

Comment:
I realize this is how it was before, but think this might have been incorrect since the settings would likely always have an owned key present. We should be treating this similar to the logic when the settings are updated, only in this case, we'd be comparing the values against the defaults.

Reply:
Updated to use the same approach for both hooks.